File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/CardEssentialsMapper.cs
using CardEssentialsSdk = OnlinePayments.Sdk.Domain.CardEssentials;
using CardEssentialsDto = Business.Domain.Payments.PaymentDetails.CardEssentials;

namespace Infrastructure.Mappers.PaymentDetails;

public static class CardEssentialsMapper
{
    public static CardEssentialsDto Map(CardEssentialsSdk? response)
    {
        return new()
        {
            CountryCode = response?.CountryCode,
            CardNumber =  response?.CardNumber,
            ExpiryDate = response?.ExpiryDate,
            Bin = response?.Bin
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/CardEssentialsMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\CardEssentials as CardEssentialsDto;
use OnlinePayments\Sdk\Domain\CardEssentials as CardEssentialsSdk;

final class CardEssentialsMapper
{
    public static function mapFromSdkResponse(?CardEssentialsSdk $response): ?CardEssentialsDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new CardEssentialsDto();
        $dto->countryCode = $response?->countryCode ?? null;
        $dto->cardNumber = $response?->cardNumber ?? null;
        $dto->expiryDate = $response?->expiryDate ?? null;
        $dto->bin = $response?->bin ?? null;

        return $dto;
    }
}
